Sources/Plasma/Apps/plClient/CMakeLists.txt

```diff
@@ -90,17 +89,106 @@ if(WIN32)
+    set(plClient_SOURCES ${plClient_SOURCES}
+        OSX/main.mm
+        OSX/PLSKeyboardEventMonitor.mm
+        OSX/PLSView.mm
+        OSX/PLSLoginWindowController.mm
+        OSX/PLSPatcherWindowController.mm
+        OSX/PLSPatcher.mm
+        OSX/PLSServerStatus.mm
+    )
+    set(plClient_HEADERS ${plClient_HEADERS}
+        OSX/PLSKeyboardEventMonitor.h
+        OSX/PLSView.h
+        OSX/PLSLoginWindowController.h
+        OSX/PLSPatcherWindowController.h
+        OSX/PLSPatcher.h
+        OSX/PLSServerStatus.h
+    )
+    set(RESOURCES 
+        OSX/MainMenu.xib
+        OSX/PLSLoginWindowController.xib
+        OSX/PLSPatcherWindowController.xib
+        OSX/banner@2x.jpeg
+        win32/res/Dirt.ICO
+    )
+    #shaders need to be compiled as part of the app
+    #this could change in the future, but for now the Metal code expects the library to be compiled in the app
+    set(plClient_SHADERS
+	../../FeatureLib/pfMetalPipeline/ShaderSrc/FixedPipelineShaders.metal
+	../../FeatureLib/pfMetalPipeline/ShaderSrc/PlateShaders.metal
+	../../FeatureLib/pfMetalPipeline/ShaderSrc/BiasNormals.metal
+	../../FeatureLib/pfMetalPipeline/ShaderSrc/CompCosines.metal
+	../../FeatureLib/pfMetalPipeline/ShaderSrc/WaveSet7.metal
+	../../FeatureLib/pfMetalPipeline/ShaderSrc/Grass.metal
+	../../FeatureLib/pfMetalPipeline/ShaderSrc/WaveDecEnv.metal
+    ../../FeatureLib/pfMetalPipeline/ShaderSrc/Avatar.metal
+    ../../FeatureLib/pfMetalPipeline/ShaderSrc/WaveDec1Lay_7.metal
+    ../../FeatureLib/pfMetalPipeline/ShaderSrc/WaveRip.metal
+    ../../FeatureLib/pfMetalPipeline/ShaderSrc/Clear.metal
+    ../../FeatureLib/pfMetalPipeline/ShaderSrc/GammaCorrection.metal
+    ../../FeatureLib/pfMetalPipeline/ShaderSrc/TextFontShader.metal
+    )
+    set_source_files_properties(${plClient_SHADERS} PROPERTIES LANGUAGE METAL)
+    source_group("Metal Shaders" FILES ${plClient_SHADERS})
+    set(plClient_SOURCES ${plClient_SOURCES} ${plClient_SHADERS})
+    include_directories("../../FeatureLib/pfMetalPipeline/metal-cpp"
+                        "../../FeatureLib/pfMetalPipeline/ShaderSrc")
+    list(APPEND plClient_RESOURCES
+        win32/res/resource.h
+        win32/res/banner.bmp
+        win32/res/Dirt.ICO
+        win32/res/headspin.ico
+    )
 else()
     list(APPEND plClient_SOURCES
         main.cpp
     )
 endif()
 
-plasma_executable(plClient WIN32 CLIENT INSTALL_PDB
-    SOURCES
-        ${plClient_SOURCES} ${plClient_HEADERS}
-        ${plClient_TEXT} ${plClient_RESOURCES}
-)
+if(APPLE)
+    plasma_executable(plClient CLIENT INSTALL_PDB SOURCES ${plClient_SOURCES} ${plClient_HEADERS} ${RESOURCES})
+    set(MAIN_NIBFILE "MainMenu")
+    set_target_properties(plClient PROPERTIES
+        BUNDLE True
+        MACOSX_BUNDLE_INFO_PLIST "${CMAKE_CURRENT_SOURCE_DIR}/OSX/Info.plist.in"
+        MACOSX_BUNDLE_GUI_IDENTIFIER org.openuru.plClient
+        MACOSX_BUNDLE_BUNDLE_VERSION "0.1"
+        MACOSX_BUNDLE_SHORT_VERSION_STRING "0.1"
```

Comment:
These (in theory) should match the values we use for the plProduct stuff: https://github.com/H-uru/Plasma/blob/2457af460fb2cd7bbc1079146e9a44505598dfb2/CMakeLists.txt#L25-L32

Reply:
MACOSX_BUNDLE_BUNDLE_VERSION should likely be some sort of build version. It's supposed to always increment so I'd have to figure out how to combine the data into a number that always goes up.

For MACOSX_BUNDLE_SHORT_VERSION_STRING - do we have an actual version number thats derived anywhere?

Reply:
https://github.com/H-uru/Plasma/blob/master/cmake/BuildInfo.cmake is used to embed git information into the executable, but that's not currently available in project mode.